android/java/org/chromium/chrome/browser/playlist/settings/BravePlaylistPreferences.java

```diff
@@ -47,36 +52,29 @@ public class BravePlaylistPreferences extends BravePreferenceFragment
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
         mEnablePlaylistSwitch = (ChromeSwitchPreference) findPreference(PREF_ENABLE_PLAYLIST);
-        mEnablePlaylistSwitch.setChecked(
-                SharedPreferencesManager.getInstance().readBoolean(PREF_ENABLE_PLAYLIST, true));
         mEnablePlaylistSwitch.setOnPreferenceChangeListener(this);
 
-        mAddToPlaylistButtonSwitch =
-                (ChromeSwitchPreference) findPreference(PREF_ADD_TO_PLAYLIST_BUTTON);
-        mAddToPlaylistButtonSwitch.setChecked(SharedPreferencesManager.getInstance().readBoolean(
-                PREF_ADD_TO_PLAYLIST_BUTTON, true));
-        mAddToPlaylistButtonSwitch.setOnPreferenceChangeListener(this);
-
-        mAutoPlaySwitch = (ChromeSwitchPreference) findPreference(PREF_AUTO_PLAY);
-        mAutoPlaySwitch.setChecked(
-                SharedPreferencesManager.getInstance().readBoolean(PREF_AUTO_PLAY, true));
-        mAutoPlaySwitch.setOnPreferenceChangeListener(this);
-
         mAutoSaveMediaForOfflinePreference =
                 (Preference) findPreference(PREF_AUTO_SAVE_MEDIA_FOR_OFFLINE);
         updateAutoSaveMedia();
 
-        mStartPlaybackSwitch = (ChromeSwitchPreference) findPreference(PREF_START_PLAYBACK);
-        mStartPlaybackSwitch.setChecked(
-                SharedPreferencesManager.getInstance().readBoolean(PREF_START_PLAYBACK, true));
-        mStartPlaybackSwitch.setOnPreferenceChangeListener(this);
+        mRememberFilePlaybackPositionSwitch =
+                (ChromeSwitchPreference) findPreference(PREF_REMEMBER_FILE_PLAYBACK_POSITION);
+        mRememberListPlaybackPositionSwitch =
+                (ChromeSwitchPreference) findPreference(PREF_REMEMBER_LIST_PLAYBACK_POSITION);
+        mContinuousListeningSwitch =
+                (ChromeSwitchPreference) findPreference(PREF_CONTINUOUS_LISTENING);
 
         mResetPlaylist = (BravePlaylistResetPreference) findPreference(PREF_RESET_PLAYLIST);
         mResetPlaylist.setOnPreferenceClickListener(preference -> {
             if (mPlaylistService != null) {
-                mPlaylistService.resetAll();
-                PlaylistPreferenceUtils.resetPlaylistPrefs(getActivity());
-                BraveRelaunchUtils.askForRelaunch(getActivity());
+                ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
let's better use what upstream has for that `PostTask`. See an example here https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/app/BraveActivity.java#L2043